[release/2.11] Generate changelog for 2.11.4

sergepetrenko opened the pull request:
Also, remove unreleased/ entries.

NO_DOC=changelog
NO_TEST=changelog
NO_CHANGELOG=changelog

Totktonada commented on changelogs/2.11.4.md:
```diff
@@ -0,0 +1,161 @@
+production for quite a while without known crashes, incorrect results or
+other showstopper bugs.
+
+This release resolves roughly 41 issues since the 2.11.3 version.
```
Seems more like 51. The tool counts changelog entries and the LuaJIT section has 2 entries, but 12 bullets.

sergepetrenko replied:
Agree. Fixed.

Totktonada commented on changelogs/2.11.4.md:
How about merging it into `Core`?

sergepetrenko replied:
Sure. Do you think we should drop the `### box` section from future changelogs altogether then?

Totktonada replied:
Maybe it worth to add a check into the `gen-release-notes` script.

Totktonada commented on changelogs/2.11.4.md:
Since there are `URI` and `YAML` subsections, we can wrap these two items to the `net.box` subsection for readability.

Totktonada commented on changelogs/2.11.4.md:
Nit: I would say `#### URI` and `#### YAML`, because these are abbreviations.

sergepetrenko replied:
Sure.

Totktonada reviewed (approved):
I've briefly glanced over the text and it seems OK. Left some minor remarks.

sergepetrenko commented:
@Totktonada thanks for the review!

PTAL.

<details><summary>Incremental diff:</summary>

```diff
diff --git a/changelogs/2.11.4.md b/changelogs/2.11.4.md
index 3a21412b61..ef983b6870 100644
--- a/changelogs/2.11.4.md
+++ b/changelogs/2.11.4.md
@@ -12,7 +12,7 @@ The label "stable" means there are 2.11.x-based applications running in
 production for quite a while without known crashes, incorrect results or
 other showstopper bugs.
 
-This release resolves roughly 41 issues since the 2.11.3 version.
+This release resolves roughly 51 issues since the 2.11.3 version.
 
 [release_policy]: https://www.tarantool.io/en/doc/latest/release/policy/
 [issues]: https://github.com/tarantool/tarantool/issues
@@ -44,6 +44,11 @@ all the new features of the 2.x series.
   already. The `fiber_join` and `fiber_join_timeout` functions now panic on a
   double join if it is possible to detect it (gh-7562).
 * Fixed hang on fiber self join (gh-10196).
+* Fixed the `wal_queue_max_size` configuration option not being applied during
+  the initial configuration (gh-10013).
+* Fixed rollback of DDL statements on the `_space` space (gh-9893).
+* Fixed memory leak on functional index drop (gh-10163).
+
 
 ### Vinyl
 
@@ -116,36 +121,30 @@ were fixed as part of this activity:
 * Fixed undefined behavior when negating `INT_MIN` integers.
 * Replaced the numeric values of NYI bytecodes that can't be compiled, with
   their names in the `jit.dump()`.
+
 Backported patches from the vanilla LuaJIT trunk (gh-10199). The following
 issues were fixed as part of this activity:
 
 * Fixed GC marking of the cdata finalizer table.
 
-### Lua
+### net.box
 
 * Fixed a corner case when the netbox connection's worker fiber could crash
   (gh-9621).
 * Fixed a bug that caused a `net.box` connection to crash after being closed
   from the connection's `on_schema_reload` trigger (gh-9621).
 
-#### Uri
+#### URI
 
 * Fixed a bug that caused characters A-F to be unsupported in IPv6
   addresses. Changed the `uri.format` output for IPv6 to be
   encapsulated in brackets `[]` (gh-9556).
 
-#### Yaml
+#### YAML
 
 * Strings with large exponential values equal to infinity are now encoded as
   strings instead of numbers (gh-10164).
 
-### Box
-
-* Fixed the `wal_queue_max_size` configuration option not being applied during
-  the initial configuration (gh-10013).
-* Fixed rollback of DDL statements on the `_space` space (gh-9893).
-* Fixed memory leak on functional index drop (gh-10163).
-
 ### Datetime
 
 * Fixed a bug with setting unspecified fields to undefined values
```

</details>

sergepetrenko commented on changelogs/2.11.4.md:
Done.

Totktonada commented:
@sergepetrenko Thanks! Seems good for me.

lenkis commented on changelogs/2.11.4.md:
```diff
@@ -0,0 +1,162 @@
+  The bug could trigger a crash in memtx while inserting a tuple into
+  a `hash` index and in vinyl while writing a bloom filter on dump or
+  compaction (gh-10090).
+* Fixed the bug that on Mac the system setting `kern.ipc.somaxconn` was ignored
+  for listening sockets. Now it is used, but capped at 32367 due to how
+  `listen()` works on Mac (gh-8130).
```
```suggestion
* Fixed a bug when the system setting `kern.ipc.somaxconn` on Mac was ignored
  for listening sockets. Now it is used, but capped at 32367 due to how
  `listen()` works on Mac (gh-8130).
```

lenkis commented on changelogs/2.11.4.md:
```diff
@@ -0,0 +1,162 @@
+  `listen()` works on Mac (gh-8130).
+* Fixed memory leaks on using dead fiber (gh-10187).
+* Fixed a possible inconsistent state entering if fibers are joined incorrectly.
+  Now the `fiber_set_joinable` function panics if the fiber is dead or joined
```
```suggestion
  Now the `fiber_set_joinable` function panics if a fiber is dead or joined
```

lenkis commented on changelogs/2.11.4.md:
```diff
@@ -0,0 +1,162 @@
+  Now the `fiber_set_joinable` function panics if the fiber is dead or joined
+  already. The `fiber_join` and `fiber_join_timeout` functions now panic on a
+  double join if it is possible to detect it (gh-7562).
+* Fixed hang on fiber self join (gh-10196).
```
```suggestion
* Fixed a hang on fiber self-join (gh-10196).
```

lenkis commented on changelogs/2.11.4.md:
```suggestion
  and an owned synchro queue could simply do nothing (gh-6860).
```

lenkis commented on changelogs/2.11.4.md:
```suggestion
* Fixed a bug that allowed the old leader in the
  `box.cfg{election_mode = 'candidate'` mode to get re-elected after resigning
  itself through `box.ctl.demote` (gh-9855).
```

sergepetrenko commented:
@lenkis, thanks for the review!

Applied and squashed your suggestions.
